### PR TITLE
test(quantic): Add QuanticCategoryFacet example

### PR DESCRIPTION
**Pull request by aelfaleh**

### CODE REVIEW

File: packages/quantic/force-app/examples/main/lwc/exampleQuanticCategoryFacet/exampleQuanticCategoryFacet.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
+import {api, LightningElement, track} from 'lwc';
+
+export default class ExampleQuanticFacet extends LightningElement {
+    @api engineId = 'quantic-category-facet-engine';
+    @track config = {};
+    isConfigured = false;
+
+    pageTitle = 'Quantic Category Facet';
+    pageDescription = 'The Quantic Category Facetis a list of values for a certain field occurring in the results, ordered using a configurable criterion.'
```
Comment (lbergeron) [inline packages/quantic/force-app/examples/main/lwc/exampleQuanticCategoryFacet/exampleQuanticCategoryFacet.js:9]: I think we should stress that the category facet is meant to be used with hierarchical values.

File: packages/quantic/force-app/examples/main/lwc/exampleQuanticCategoryFacet/exampleQuanticCategoryFacet.js [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        },
+        {
+            attribute: 'label',
+            label: 'Label',
+            description: 'The label to use as the facet title.',
+            defaultValue: 'Country'
+        },
+        {
+            attribute: 'basePath',
+            label: 'BasePath',
```
Comment (lbergeron) [inline packages/quantic/force-app/examples/main/lwc/exampleQuanticCategoryFacet/exampleQuanticCategoryFacet.js:25]: I think sentence case is better for the labels. Please update other labels in this file as well.

`BasePath` --> `Base path`

File: packages/quantic/force-app/examples/main/lwc/exampleQuanticCategoryFacet/exampleQuanticCategoryFacet.js [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+        {
+            attribute: 'noFilterByBasePath',
+            label: 'No Filter By Base Path',
+            description: 'Whether not to use the `basePath` as a filter for the results.',
+            defaultValue: false
+        },
+        {
+            attribute: 'noFilterFacetCount',
+            label: 'No Filter Facet Count',
+            description: 'Whether not to exclude the parents of folded results when estimating the result count for each facet value.',
```
Comment (lbergeron) [inline packages/quantic/force-app/examples/main/lwc/exampleQuanticCategoryFacet/exampleQuanticCategoryFacet.js:38]: Maybe it is just me, but I have a hard time understanding what should happen when setting this option :sweat_smile:

The `Whether not to exclude...` part is difficult to understand, and the relation between `filter` and `parents of folded results` is unclear.

Using `Whether to include the parents...` would help make the description clearer, even though the description still seem unrelated to the attribute name.

File: packages/quantic/force-app/examples/main/lwc/exampleQuanticCategoryFacet/exampleQuanticCategoryFacet.js [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+            defaultValue: false
+        },
+        {
+            attribute: 'delimitingCharacter',
+            label: 'Delimiting Character',
+            description: 'The character that separates the values of the target multi-value field.',
```
Comment (lbergeron) [inline packages/quantic/force-app/examples/main/lwc/exampleQuanticCategoryFacet/exampleQuanticCategoryFacet.js:44]: The actual description could be misleading. It depends on how the field has been configured in the index.

If the field is created as a **hierarchical field**, the default level delimiter is `;`, but mentioning `multi-value field` in the description is kind of misleading.

If the field is created as a **multi-value field**, the default multi-value delimiter is `;`, and the default level delimiter is `>`. In this case, the multi-value delimiter is handled internally by the platform, so it is transparent to the Search API consumers.

Bottom line, I think it would make things clearer to use "level delimiter" in the attribute description.

File: packages/quantic/force-app/examples/main/lwc/exampleQuanticCategoryFacet/exampleQuanticCategoryFacet.js (hunk 1 above)
Comment (nathanlb) [reply packages/quantic/force-app/examples/main/lwc/exampleQuanticCategoryFacet/exampleQuanticCategoryFacet.js:9]:
```suggestion
    pageDescription = 'The Quantic Category Facet is a list of values for a certain field occurring in the results displayed in a browsable, hierarchical fashion and ordered using a configurable criterion.'
```
